src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -445,11 +448,124 @@ public DatabaseRecordActions(DocumentDatabase database, Logger log)
+                    {
+                        pullReplication.Disabled = true;
+                        tasks.Add(_database.ServerStore.SendToLeaderAsync(new UpdatePullReplicationAsHubCommand(_database.Name)
+                            {
+                                Definition = pullReplication
+                            }
+                        ));
+                    }
+                    progress.HubPullReplicationsUpdated = true;
+                }
+
+                if (currentDatabaseRecord?.Sorters.Count == 0 &&
+                    databaseRecord?.Sorters.Count > 0)
+                {
+                    if (_log.IsInfoEnabled)
+                        _log.Info("Configuring sorters configuration from smuggler");
+
+                    tasks.Add(_database.ServerStore.SendToLeaderAsync(new PutSortersCommand(_database.Name)
+                    {
+                        Sorters = databaseRecord.Sorters.Values.ToList()
+                    }));
+
+                    progress.SortersUpdated = true;
+                }
+
+                if (currentDatabaseRecord?.ExternalReplications.Count == 0 &&
+                    databaseRecord?.ExternalReplications.Count > 0)
+                {
+                    if (_log.IsInfoEnabled)
+                        _log.Info("Configuring external replications configuration from smuggler");
+                    foreach (var replication in databaseRecord.ExternalReplications)
+                    {
+                        replication.Disabled = true;
+                        tasks.Add(_database.ServerStore.SendToLeaderAsync(new UpdateExternalReplicationCommand(_database.Name)
+                        {
+                            Watcher = replication
+                        }));
+                    }
+                    progress.ExternalReplicationsUpdated = true;
+                }
+
+                if (currentDatabaseRecord?.RavenEtls.Count == 0 &&
+                    databaseRecord?.RavenEtls.Count > 0)
+                {
+                    if (_log.IsInfoEnabled)
+                        _log.Info("Configuring raven etls configuration from smuggler");
+                    foreach (var etl in databaseRecord.RavenEtls)
+                    {
+                        etl.Disabled = true;
+                        tasks.Add(_database.ServerStore.SendToLeaderAsync(new AddRavenEtlCommand(etl, _database.Name)));
+                    }
+                    progress.RavenEtlsUpdated = true;
+                }
+
+                if (currentDatabaseRecord?.SqlEtls.Count == 0 &&
+                    databaseRecord?.SqlEtls.Count > 0)
+                {
+                    if (_log.IsInfoEnabled)
+                        _log.Info("Configuring sql etls configuration from smuggler");
+                    foreach (var etl in databaseRecord.SqlEtls)
+                    {
+                        tasks.Add(_database.ServerStore.SendToLeaderAsync(new AddSqlEtlCommand(etl, _database.Name)));
```

Comment:
Why are you calling the commands here and not modifying the database record directly?

Reply:
I need to get new information for the command. like, TaskId, mentor node etc.